tests/e2e/performance/csi_tests/test_pvc_snapshot_performance.py

```diff
@@ -509,12 +630,63 @@ def test_pvc_snapshot_performance_multiple_files(
         log.info("Deleting the elastic-search instance")
         self.es.cleanup()
 
+        avg_c_time = statistics.mean(all_results)
+        t_dateset = int(data_set / 3)
+
         log.info(f"Full test report for {interface}:")
         log.info(
             f"Test ran {self.tests_numbers} times, " f"All results are {all_results}"
         )
-        log.info(f"The average creation time is : {statistics.mean(all_results)}")
+        log.info(f"The average creation time is : {avg_c_time}")
         log.info(
             f"Number of Files on the volume : {total_files:,}, "
-            f"Total dataset : {int(data_set / 3)} GiB"
+            f"Total dataset : {t_dateset} GiB"
         )
+
+        full_results.add_key("avg_snapshot_creation_time_insecs", avg_c_time)
+        full_results.all_results["total_files"] = total_files
+        full_results.all_results["total_dataset"] = t_dateset
+        full_results.all_results["creation_time"] = all_results
+
+        # Write the test results into the ES server
+        log.info("writing to elastic search")
+        full_results.es_write()
+        res_link = full_results.results_link()
+        # write the ES link to the test results in the test log.
+        logging.info(f"The result can be found at : {res_link}")
+
+        # Create text file with results of all subtest
+        self.write_result_to_file(res_link)
+
+    def test_pvc_snapshot_performance_multiple_file_results(self):
```

Comment:
In this test you 've added 2 results functions: test_pvc_snapshot_performance_multiple_file_results and test_pvc_snapshot_performance_results. 

Those results should be pushed together to the dashboard in one function, in order to avoid code duplication. 
You are welcome to check the example in  def test_bulk_pvc_creation_deletion_performance_results(self) in the test_pvc_bulk_creation_performance.py test.

Reply:
Yeah better add one function something like https://github.com/red-hat-storage/ocs-ci/blob/master/tests/e2e/performance/io_workload/test_fio_benchmark.py#L598